tests/unit/test_gateway.py

```diff
@@ -0,0 +1,42 @@
+from types import GeneratorType
+
+import pytest
+
+from exodus_gw.models import Publish
+from exodus_gw.routers import gateway
+
+
+def test_healthcheck():
+    assert gateway.healthcheck() == {"detail": "exodus-gw is running"}
+
+
+@pytest.mark.parametrize(
+    "env",
+    [
+        "dev",
+        "qa",
+        "stage",
+        "prod",
+    ],
+)
+def test_publish_env_exists(env, mock_db_session):
+    publish = gateway.publish(env=env, db=mock_db_session)
```

Comment:
All of these `gateway.publish` calls fail because you aren't awaiting the coroutine.
```suggestion
    publish = await gateway.publish(env=env, db=mock_db_session)
```
^ should fix it.